include/swift/Option/Options.td

```diff
@@ -60,6 +60,9 @@ def SupplementaryOutput : OptionFlag;
 // The option should be accepted by swift-symbolgraph-extract.
 def SwiftSymbolGraphExtractOption : OptionFlag;
 
+// The option should be accepted by swift-synthesize-interface.
+def SwiftSynthesizeInterfaceOption : OptionFlag;
```

Comment:
It would be ideal if we could unify `InterfaceSubContextDelegateImpl::inheritOptionsForBuildingInterface` with something like this. 

Because right now if these two get out-of-sync, I imagine this tool may produce outputs that would differ from what the compiler would actually do during compilation.

Reply:
I might be missing the connection here, since the primary goal for this mode is to print the synthesized Swift decls for a C/Obj-C/C++ module. What's the failure case you're thinking of? Someone using this with a `.swiftinterface` file on the search path and having it output something different than the interface that would result from compiling it, someone seeing a mismatch between the textual interface and the synthesized interface of an already compiled `.swiftmodule`, or something else?

Reply:
I think my mind went in the direction of using a tool like this for Swift code. 
It is much less of a concern for C/Obj-C/C++ modules, you're correct. 

> Someone using this with a .swiftinterface file on the search path and having it output something different than the interface that would result from compiling it, someone seeing a mismatch between the textual interface and the synthesized interface of an already compiled .swiftmodule, or something else?

These are the kinds of failure scenarios I was thinking of, yes.

Reply:
> I think my mind went in the direction of using a tool like this for Swift code.
> It is much less of a concern for C/Obj-C/C++ modules, you're correct.

Since this tool has a similar interface to `swift-symbolgraph-extract`, it doesn't take any Swift source files as inputs; it just takes the command line arguments needed to set up the search paths of the compiler instance, and then the module specified by `-module-name` will be loaded through the standard mechanisms.

However, using it to dump the interface of an already compiled `.swiftmodule` would still be useful. I tested a couple of the cases you were concerned about, and they fail with the expected diagnostics. For example, if I compile a `.swiftinterface` and `.swiftmodule` for iOS and then try to run `swift-synthesize-interface` with `-target arm64-apple-macosx14.0` using either the binary module or the textual interface on the search path, I get the following error:

```
<unknown>:0: error: module 'Module' was created for incompatible target arm64-apple-ios15.0-simulator
```

Does that address your concerns?